storage/range.go

```diff
@@ -278,20 +279,76 @@ func (r *Range) IsFirstRange() bool {
 	return bytes.Equal(r.Desc().StartKey, engine.KeyMin)
 }
 
-// IsLeader returns true if this range replica is the raft leader.
-// TODO(spencer): this is always true for now.
-func (r *Range) IsLeader() bool {
-	return true
+// getLease returns the current leader lease.
+func (r *Range) getLease() *proto.Lease {
+	return (*proto.Lease)(atomic.LoadPointer(&r.lease))
+}
+
+// HasLeaderLease returns true if this range replica currently holds
+// the Raft leader lease. It's possible that the replica is not in
+// fact the leader, but holding the lease means that no other replica
```

Comment:
FWIW, after we move to two-phase elections it should no longer be possible for a replica to lose leadership while holding the lease (although being the raft leader is a fairly weak claim, since raft doesn't guarantee that there is only one leader. Instead, it guarantees that at most one leader is able to make progress). It is possible that other nodes will be "leaders" while this node has the lease (partitioned raft leaders do not lose their leadership until they regain contact with the cluster), but if this node has the lease then all other leases have expired and these other "leaders" will be unable to regain them.

I think we may need to return the term of the lease here, and add a leaseTerm argument to MultiRaft.SubmitCommand.